internal/configs/provider.go

```diff
@@ -13,17 +13,14 @@ import (
 	"github.com/hashicorp/hcl/v2/hclsyntax"
 
 	"github.com/opentofu/opentofu/internal/addrs"
+	"github.com/opentofu/opentofu/internal/evalchecks"
+	"github.com/opentofu/opentofu/internal/instances"
 	"github.com/opentofu/opentofu/internal/tfdiags"
 )
 
-// Provider represents a "provider" block in a module or file. A provider
-// block is a provider configuration, and there can be zero or more
-// configurations for each actual provider.
-type Provider struct {
-	Name       string
-	NameRange  hcl.Range
-	Alias      string
-	AliasRange *hcl.Range // nil if no alias set
+type ProviderCommon struct {
```

Comment:
I understand, we already discussed this previously and I do not have a strong opinion here.

But...

Do we need `ProviderCommon`? It is not useful by itself and doesn't say anything about what it really is.

We could move `ProviderCommon` fields to `ProviderBlock` and tweak a `Provider` struct to include a `ConfigBlock` field of type `ProviderBlock` (instead of embedding). This looks good to me since we have both clean separation and source configuration values.

Reply:
i would tend to agree, i'm not a fan of embedding. But i'm ok with what we have